### PR TITLE
Add new reverted-transactions agent

### PR DESCRIPTION
**Pull request by skhomuti**

### CODE REVIEW

File: reverted-tx-watcher/src/agent-reverted-tx.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  for (let [name, address] of addresses) {
+    if (address in txEvent.addresses) {
+      const receipt = await ethersProvider.getTransactionReceipt(
+        txEvent.transaction.hash
+      );
+      if (receipt.status === 0) {
+        const fromSelf = address === txEvent.from;
+        findings.push(
+          Finding.fromObject({
+            name: "Reverted tx",
```
Comment (dgusakov) [inline reverted-tx-watcher/src/agent-reverted-tx.ts:89]:
```suggestion
            name: "🤔 Reverted TX detected",
```

File: reverted-tx-watcher/src/agent-reverted-tx.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            description:
+              `Reverted tx ${fromSelf ? "from" : "to"} the ${name} contract. ` +
```
Comment (dgusakov) [inline reverted-tx-watcher/src/agent-reverted-tx.ts:91]:
```suggestion
              `Reverted TX ${fromSelf ? "from" : "to"} the ${name} contract. ` +
```

File: reverted-tx-watcher/src/agent-reverted-tx.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    stakingRouter: await locator.stakingRouter(),
+    treasury: await locator.treasury(),
+    validatorsExitBusOracle: await locator.validatorsExitBusOracle(),
+    withdrawalQueue: await locator.withdrawalQueue(),
+    withdrawalVault: await locator.withdrawalVault(),
+    oracleDaemonConfig: await locator.oracleDaemonConfig(),
+  };
+
+  addresses = Object.entries(contracts);
+  addresses.push(["EasyTrack", EASY_TRACK_ADDRESS]);
```
Comment (dgusakov) [inline reverted-tx-watcher/src/agent-reverted-tx.ts:61]: Consider adding other Lido contracts not included in the LidoLocator

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: reverted-tx-watcher/src/constants.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// COMMON CONSTS
+
+// ADDRESSES AND EVENTS
+
+export const LIDO_LOCATOR_ADDRESS =
+  "0xc1d0b3de6792bf6b4b37eccdcc24e45978cfd2eb"; // should be lowercase
```
Comment (madlabman) [inline reverted-tx-watcher/src/constants.ts:6]: Why lowercased?

Comment (skhomuti) [reply reverted-tx-watcher/src/constants.ts:6]: copypasta from demo bot

File: reverted-tx-watcher/src/agent-reverted-tx.ts (hunk 3 above)
Comment (skhomuti) [reply reverted-tx-watcher/src/agent-reverted-tx.ts:61]: wanted to ask you what else contracts needed to be added

Comment (dgusakov) [reply reverted-tx-watcher/src/agent-reverted-tx.ts:61]: All major from the docs IMO

Comment (skhomuti) [reply reverted-tx-watcher/src/agent-reverted-tx.ts]: added some

File: reverted-tx-watcher/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Lido bot detecting reverted transactions
+
+## Supported chains
+
+- Ethereum mainnet
+
+## Alerts
+
```
[the remark this answers is not in the document]
Comment (skhomuti) [reply reverted-tx-watcher/README.md:8]: added

**Review by dgusakov — APPROVED** [review]

**Pull request merged**